lib/Client/AbstractClient.php

```diff
@@ -20,10 +23,19 @@
      */
     private $options;
 
-    public function __construct(array $options = [])
+    /**
+     * @var ResponseFactoryInterface|ResponseFactory
+     */
+    protected $responseFactory;
+
+    public function __construct(array $options = [], $responseFactory = null)
     {
         $this->options = new ParameterBag();
         $this->options = $this->doValidateOptions($options);
+        if (!$responseFactory instanceof ResponseFactoryInterface && !$responseFactory instanceof ResponseFactory) {
+            $responseFactory = new MessageFactory();
```

Comment:
Rather throw an exception then silently correct the mistake.

Reply:
Should I remove the message factory completely or just create it if `$responseFactory` is null?
I was thinking to create it so we don't break BC

Reply:
Yeah, We do want to avoid BC breaks. 

If `$responseFactory` is null then we add the `$requestFactory = new MessageFactory();`. If it is not null and if it is not an instance of HTTPlug message factory or Iterop, then we have to throw an exception